src/ripple/basics/base_uint.h

```diff
@@ -320,6 +320,7 @@ class base_uint
     {
         unsigned char* pOut  = begin ();
 
+        assert(sizeof(pn) == bytes);
```

Comment:
Could `static_assert` instead (tho make sure you test on windows, I only tested linux).